storm-core/src/jvm/backtype/storm/utils/RateTracker.java

```diff
@@ -72,7 +80,7 @@ public RateTracker(int validTimeWindowInMils, int numOfSlides, boolean simulate
      * @param count number of arrivals
      */
     public void notify(long count) {
-        _histograms[_histograms.length-1]+=count;
+        _histograms[_numOfSlides - 1] += count;
```

Comment:
@revans2 
Thanks for your comment. Will you solve [STORM-1078] by yourself? I can solve this bug if you are too busy to do it.